server/aws-lsp-codewhisperer/src/shared/codeWhispererService.ts

```diff
@@ -138,6 +142,23 @@ export abstract class CodeWhispererServiceBase {
 
     abstract generateSuggestions(request: GenerateSuggestionsRequest): Promise<GenerateSuggestionsResponse>
 
+    abstract constructSupplementalContext(
```

Comment:
since it will be reused by both "inlineCompletionHandler" and "editCompletionHandler", move to here such that we don't have duplicate

Reply:
Also it can leverage the "IAM" vs. "Token" subclass to have their own implementation. 
Currently we don't have supplemental context for IAM and only Token based has such.